freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UmsUsersStateProvider.java

```diff
@@ -120,7 +122,8 @@ private WorkloadCredential getCredentials(String userCrn, Optional<String> reque
         GetActorWorkloadCredentialsResponse response = grpcUmsClient.getActorWorkloadCredentials(IAM_INTERNAL_ACTOR_CRN, userCrn, requestId);
         String hashedPassword = response.getPasswordHash();
         List<ActorKerberosKey> keys = response.getKerberosKeysList();
-        return new WorkloadCredential(hashedPassword, keys);
+        long expirationDate = response.getPasswordHashExpirationDate();
```

Comment:
Suggestion: convert it to an Instant here.

Reply:
converted it to an Optional<Instant> so we can do the 0-checking here where the 0 is meaningful.